_posts/2016-10-16-getting-started-in-scala-on-android.md

```diff
@@ -0,0 +1,217 @@
+Alternatively, you can install [Android Studio](https://developer.android.com/studio/index.html), which comes with the Android SDK and emulators.
+
+###Set the ANDROID_HOME environment variable
+On Mac OSX/Linux, you can just export the variable
+
+```bash
+$ export ANDROID_HOME=path_to_your_android_sdk
+```
+
+or add it to your bash_rc or bash_profile.
+
+NOTE: On Mac OSX, if Android Studio is installed, the Android SDK is usually located at `/Users/your_user_name/Library/Android/sdk/`
+
+##Add SBT plugin
+The easiest way to install the Android SDK Plugin for SBT is to do it globally. For this, you'll need to create a file in the SBT plugins folder:
+
+```bash
+~/.sbt/0.13/plugins/android.sbt
+```
+
+and add the following line:
+
+```bash
+addSbtPlugin("org.scala-android" % "sbt-android" % "1.7.0")
+```
+
+##Create the Android project
+Let's create a folder for our Android project:
+
+```bash
+$ mkdir my-project
+$ cd my-project
+```
+
+Now, we are going to use the SBT plugin to create a template project. First, run SBT:
+
+```bash
+$ sbt
+```
+
+Then, use the plugin to create the project.
+
+```bash
+> gen-android <package_name> <project_name>
+```
+
+For example:
+
+```bash
+> gen-android com.codurance scala_on_android
+```
+
+The project structure created looks like this:
+
+```text
+my-project/
+|-- project/
+|   |-- android.sbt
+|   |-- build.properties
+|-- src/
+|   |-- androidTest/
+|       |-- java/
+|           |-- com/
+|               |-- codurance/
+|                   |-- Junit3MainActivityTest.java
+|                   |-- Junit4MainActivityTest.java
+|   |-- main/
+|       |-- res/
+|           // Android resorces folders
+|       |-- scala/
+|           |-- com/
+|               |-- codurance/
+|                   |-- MainActivity.scala
+|       |-- AndroidManifest.xml
+|-- build.sbt
+|-- lint.xml
+```
+
+The SBT plugin creates a project structure with the minimum files needed to run an Android project, plus a setup for running instrumentation tests. Notice that the test classes generated are in Java, and the Main Activity is in Scala.
+
+The most interesting file is build.sbt. I've added some comments to explain what's the purpose of each line.
+
+```java
+// Version of the Scala runtime
+scalaVersion := "2.11.8"
+
+// Use the Android plugin
+enablePlugins(AndroidApp)
+// Add support for vector drawables
+useSupportVectors
+
+// Android version code (Same as versionCode on Gradle projects)
+versionCode := Some(1)
+// Android version name (Same as versionName on Gradle projects)
+version := "0.1-SNAPSHOT"
+
+// Instrumentation tests runner (Same as testInstrumentationRunner on Gradle projects)
+instrumentTestRunner :=
+  "android.support.test.runner.AndroidJUnitRunner"
+
+// Android platform target (Same as targetSdkVersion on Gradle projects)
+platformTarget := "android-24"
+
+// Java compile options
+javacOptions in Compile ++= "-source" :: "1.7" :: "-target" :: "1.7" :: Nil
+
+// Libraries
+libraryDependencies ++=
```

Comment:
This is a bit confusing, it appends all dependencies, no matter whether they are for testing or production. It differentiates between them using `% "androidTest"` for the testing ones.
This is how the plugin generates the dependencies. From this, I'd probably separate them to make it more explicit. I wanted to keep the original file to explain what a new user would find.
The same problem exists on Gradle, you usually add all dependencies in the same section and only `compile`, `testCompile` and `androidTestCompile` help you differentiate between them, although it's more readable because those words are at the beginning of the line, not at the end as in the code generated by this plugin.